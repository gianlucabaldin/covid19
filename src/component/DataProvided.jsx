import React from 'react';
import { Link, Typography } from '@material-ui/core';
import { useTranslation } from 'react-i18next';

const DataProvided = ({ href }) => {
  const { t } = useTranslation();
  return (
    <Typography style={{ fontStyle: 'italic', marginRight: 50 }}>
      {t('dataprovided.data-provided-by')} <Link href={href}>{href}</Link>
    </Typography>
  );
};

export default DataProvided;
