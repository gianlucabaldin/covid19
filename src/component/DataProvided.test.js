import React from 'react';
import { shallow } from 'enzyme';
import DataProvided from './DataProvided';

describe('DataProvided component', () => {
  let wrapper = shallow(<DataProvided />);
  it('should renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render date and ref', () => {
    wrapper = shallow(<DataProvided href="www.google.com" />);
    expect(wrapper).toIncludeText('Data provided by www.google.com');
  });
});
