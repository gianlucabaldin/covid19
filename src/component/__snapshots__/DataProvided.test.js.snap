// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataProvided component should renders properly 1`] = `
<WithStyles(ForwardRef(Typography))
  style={
    Object {
      "fontStyle": "italic",
      "marginRight": 50,
    }
  }
>
  Data provided by 
  <WithStyles(ForwardRef(Link)) />
</WithStyles(ForwardRef(Typography))>
`;
